### DISCUSSION TITLE
Space in NuGet.config packageSource looks to be encoded as __x0020__ instead of _x0020_

### DISCUSSION THREAD

**Discussion by ryankanno** [renovatebot/renovate#27082] [Request Help]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us what version of Renovate you run.

37.170.0

### If you're self-hosting Renovate, select which platform you are using.

Azure DevOps (dev.azure.com)

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the problem

 We have a custom NuGet.config that lists a private artifact feed in packageSources.

```
<packageSources>
    <add key="Foobar Azure Artifact Feed" value="https://pkgs.dev.azure.com/foobar/barfoo/_packaging/barfoo/nuget/v3/index.json" />
</packageSources>
``` 

When there is a space in the key, according to the [NuGet config file reference](https://learn.microsoft.com/en-us/nuget/reference/nuget-config-file#packagesourcecredentials), this should be encoded as \_x0020\_. It looks like this is being encoded with \_\_x0020\_\_ which ultimately prevents the private artifact feed from being authenticated/queried.

This was manifesting itself in the logs as an error NU301: Unable to load service index for source https://pkgs.dev.azure.com/foobar/barfoo/_packaging/barfoo/nuget/v3/index.json when Renovate was attempting to restore the projects. Ultimately, this would lead to "Artifact update problems" in the Renovate PR; packages.lock.json update failures with a warning not to merge the PR.

Initially, I assumed it was the RENOVATE_HOST_RULES, but according to the debug logs, they were being read fine.

```
DEBUG: Env config
       "config": {
         "hostRules": [
           {
             "hostType": "nuget",
             "matchHost": "pkgs.dev.azure.com",
             "username": "foobar",
             "password": "***********"
           },
           {"hostType": "nuget", "matchHost": "nuget.org", "enabled": "false"}
         ]
       }
```
Based on the NU301 error, I then assumed the credentials were wrong / PAT had incorrect permissions, but they were fine as I could curl the index with the PAT via the -u flag. Ultimately, I ended up running the Renovate docker container and mounting /tmp to the host machine so that I could take a peek at the generated nuget file that was being passed to the following dotnet restore command.

```
"cmd": "/bin/sh -c dotnet restore Foobar.csproj --force-evaluate --configfile /tmp/renovate/cache/__renovate-private-cache/nuget/nuget.config",
```

It was upon looking at this nuget.config where I noticed what the Foobar Azure Artifact Feed looked like:

```
<?xml version="1.0" encoding="utf-8"?>
<configuration>
  <packageSources>
    <add key="Foobar Azure Artifact Feed" value="https://pkgs.dev.azure.com/foobar/barfoo/_packaging/barfoo/nuget/v3/index.json" protocolVersion="3" />
  </packageSources>
  <packageSourceCredentials>
    <Foobar__x0020__Azure__x0020__Artifact__x0020__Feed>
      <add key="Username" value="<correct_value_from_host_rules>" />
      <add key="ClearTextPassword" value="<correct_value_from_host_rules>" />
      <add key="ValidAuthenticationTypes" value="basic" />
    </Foobar__x0020__Azure__x0020__Artifact__x0020__Feed>
  </packageSourceCredentials>
</configuration>
```

I took this nuget.config file and ran the exact dotnet restore command being run in the Renovate container on my local machine (but added a --no-cache flag). Like Renovate, I also was unable to restore the project due to the same NU301 error. Once the extra _ were removed, running the dotnet restore command on the host machine worked.

I also confirmed that if I changed the name of the original NuGet.config to:

```
<packageSources>
    <add key="foobar-artifact-feed" value="https://pkgs.dev.azure.com/foobar/barfoo/_packaging/barfoo/nuget/v3/index.json" />
</packageSources>
``` 

I would no longer get the NU1301 and am able to restore / updated packages.lock.json files. Here's the [line](https://github.com/renovatebot/renovate/blob/ca80f6c2ed45ff0e9e163b0a6692603ac6881c5e/lib/modules/manager/nuget/config-formatter.ts#L125) which looks to be doing the encoding. 

I took a peek at what the NuGet tools do, and it [looks](https://github.com/NuGet/NuGet.Client/blob/2fc1ae40cecf6f1a376f76134fd768e0bc4e3372/src/NuGet.Core/NuGet.Configuration/Settings/Items/CredentialsItem.cs#L32) like they are ultimately calling XmlUtility.EncodeLocalName which would return a single _.

### Relevant debug logs

<details><summary>Logs</summary>

```

Copy/paste the relevant log(s) here, between the starting and ending backticks

```

</details>

### Have you created a minimal reproduction repository?

I have explained in the description why a minimal reproduction is impossible

**Comment by viceice** [comment]
it's a bug, feel free to open a PR to fix. 🤗
